### PR TITLE
Exposing pipeline metrics

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: quickwit/quickwit-cluster/src/cluster.rs [hunk 1]
```diff
@@ -301,6 +301,24 @@ impl Cluster {
         tokio::time::sleep(GOSSIP_INTERVAL * 2).await;
     }
 
+    /// This exposes in chitchat some metric about the cpu usage of cooperative pipelines.
```
Comment (guilload) [inline quickwit/quickwit-cluster/src/cluster.rs:304]:
```suggestion
    /// This exposes in chitchat some metrics about the CPU usage of cooperative pipelines.
```

File: quickwit/quickwit-cluster/src/cluster.rs [hunk 2]
```diff
@@ -301,6 +301,24 @@ impl Cluster {
...
+    /// The metric are exposed as follows:
```
Comment (guilload) [inline quickwit/quickwit-cluster/src/cluster.rs:305]:
```suggestion
    /// The metrics are exposed as follows:
```

File: quickwit/quickwit-cluster/src/cluster.rs [hunk 3]
```diff
@@ -301,6 +301,24 @@ impl Cluster {
...
+    /// Key:        pipeline_metrics:<index_uid>:<source_id>
+    /// Value:      79‰,76MB/s
```
Comment (guilload) [inline quickwit/quickwit-cluster/src/cluster.rs:307]: Just so it's clear that we're dealing with thousands.
```suggestion
    /// Value:      179‰,76MB/s
```

File: quickwit/quickwit-proto/src/lib.rs [hunk 4]
```diff
@@ -252,3 +252,19 @@ impl search::SortOrder {
 }
 
 impl quickwit_common::pubsub::Event for ReportSplitsRequest {}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize)]
```
Comment (guilload) [inline quickwit/quickwit-proto/src/lib.rs:256]: Let's move that to ` quickwit_proto::indexing`.

File: quickwit/quickwit-proto/src/lib.rs [hunk 5]
```diff
@@ -252,3 +252,19 @@ impl search::SortOrder {
...
+pub struct PipelineMetrics {
+    pub cpu_thousandth: u16,
```
Comment (guilload) [inline quickwit/quickwit-proto/src/lib.rs:258]: Looking at other codebases (Systemd, Linux Kernel, Kube), looks like `cpu_shares` is more commonly used with the `millicore` unit.

**Review by guilload — APPROVED** [review]

File: quickwit/quickwit-proto/src/lib.rs (hunk 5 above)
Comment (fulmicoton) [reply quickwit/quickwit-proto/src/lib.rs:258]: cool i did not know that word!

File: quickwit/quickwit-proto/src/lib.rs (hunk 4 above)
Comment (fulmicoton) [reply quickwit/quickwit-proto/src/lib.rs]: done

**Pull request merged**